forem/forem
# Implement Unified Embeds for Github Repo, Issue, Pull-Request and Comment URLs

## What type of PR is this? (check all applicable)


## Description
This PR implements the [unified embed tag](https://github.com/forem/forem/issues/15099) for Github repository, issue, pull-request and comment embeds.

## Related Tickets & Documents
closes https://github.com/forem/forem/issues/15557
closes https://github.com/forem/forem/issues/15558

## QA Instructions, Screenshots, Recordings
Example GitHub URLs:
```
https://github.com/forem/forem
https://github.com/forem/forem/issues/15099
https://github.com/forem/forem/pull/16415
https://github.com/forem/forem/pull/16378#issuecomment-1026386485
```
_(the following two URLs are not handled by GitHub's Issue-Comments API, so the embed should fall back to the PR)_
```
https://github.com/forem/forem/pull/16378#discussion_r796201397
https://github.com/forem/forem/pull/16378#pullrequestreview-869248370
```

Old Embed examples:
```
{% github forem/forem %}
{% github forem/forem no-readme %}
{% github https://github.com/forem/forem/issues/9 %}
```

- As a user, create GitHub liquid tag using this format: {% embed <github-url> %}. The Liquid Tag should render properly.
- Also create liquid tags using the old method (see examples above). This should also work, as I am leaving the old implementation in place for now.


### UI accessibility concerns?
None

## Added/updated tests?

## [Forem core team only] How will this change be communicated?
Considering the project as a whole, I shall update the Liquid Tag documentation at the end. Since I am not changing the current implementation, the present documentation is still valid. Plus, I will avoid updating the docs piecemeal, and risk confusing users.

## [optional] Are there any post deployment tasks we need to perform?
None
